src/vision/status_quo/aws_engineer/figuring_out_the_best_option.md

```diff
@@ -0,0 +1,3 @@
+# Status quo of an AWS engineer: Figuring out the best option
+
+Sometimes after working on `AsyncRead`, Alan stumbles over the `async-trait` crate. This crate offers a macro that will let him add `async fn` to traits. He's excited about this because it seems like it would allow him to rewrite some of the custom `AsyncRead` impls in a cleaner way. The only problem is that he can't really judge what the implications are going to be -- will it be faster? Slower? It's hard to tell until it's done. He feels like this comes up a lot in Rust: he is forced to make a choice and see it all the way through to the end before he can decide whether he likes it (or if it will work at all: sometimes he encounters a compiler error part of the way through that he just can't figure out how to resolve). It's particularly frustrating in Async Rust where [there seem to be so many options to choose from](https://rust-lang.github.io/wg-async-foundations/vision/status_quo/barbara_needs_async_helpers.html).
```

Comment:
```suggestion
Sometime after working on `AsyncRead`, Alan stumbles over the `async-trait` crate. This crate offers a macro that will let him add `async fn` to traits. He's excited about this because it seems like it would allow him to rewrite some of the custom `AsyncRead` impls in a cleaner way. The only problem is that he can't really judge what the implications are going to be -- will it be faster? Slower? It's hard to tell until it's done. He feels like this comes up a lot in Rust: he is forced to make a choice and see it all the way through to the end before he can decide whether he likes it (or if it will work at all: sometimes he encounters a compiler error part of the way through that he just can't figure out how to resolve). It's particularly frustrating in Async Rust where [there seem to be so many options to choose from](https://rust-lang.github.io/wg-async-foundations/vision/status_quo/barbara_needs_async_helpers.html).
```